assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/SpecificationFactory.java

```diff
@@ -109,4 +118,49 @@ public static Specification isBetween(String attribute, int min, int max) {
   public static Specification isBetween(String attribute, double min, double max) {
     return (root, query, cb) -> cb.between(root.get(attribute), min, max);
   }
+
+  /**
+   * Returns a list of values from the provided column filter.
+   * Correctly handles assessment LocalDate fields, including dot-notated sub entity fields.
+   *
+   * @param cf The column filter to use
+   * @return A list of values
+   */
+  public static List<Object> getDateAwareValuesFromColumnFilter(ColumnFilter cf) {
+    List<Object> valuesList;
+    try {
+      String[] subEntity = StringUtils.split(cf.getName(), DOT);
+      Field field;
+      if (subEntity.length == 1) {
+        field = Assessment.class.getDeclaredField(subEntity[0]);
+      } else {
+        switch (subEntity[0].toLowerCase()) {
+          case "revalidation":
+            field = Revalidation.class.getDeclaredField(subEntity[1]);
+            break;
+          case "outcome":
+            field = AssessmentOutcome.class.getDeclaredField(subEntity[1]);
+            break;
+          case "detail":
+            field = AssessmentDetail.class.getDeclaredField(subEntity[1]);
+            break;
+          default:
+            throw new NoSuchFieldException(subEntity[0]);
+        }
+      }
+      if (field.getType().equals(LocalDate.class)) {
+        //dates need to be handled differently to strings / numbers
+        valuesList = cf.getValues().stream()
+            .map(c -> LocalDate.parse(c.toString()))
+            .collect(Collectors.toList());
+      } else {
+        valuesList = cf.getValues();
+      }
+    } catch (NoSuchFieldException e) {
+      valuesList = cf.getValues(); //allow calling function to handle this
+    } catch (DateTimeParseException e) {
+      valuesList = null; //query will thus ignore invalid date values
```

Comment:
This is debatable: if it simply returns the original string values this will cause an obvious error in the UI, which is possibly better than silently swallowing bad date criteria. The original functionality would have simply crashed out on any date queries, so its not 100% clear what would be best here.